packages/bigtable/src/table.js

```diff
@@ -735,20 +735,51 @@ Table.prototype.getRows = function(options, callback) {
  * table.insert(entries, callback);
  *
  * //-
- * // If you don't provide a callback, an EventEmitter is returned. Listen for
- * // the error event to catch API and insert errors, and complete for when
- * // the API request has completed.
+ * // If the callback is omitted, we'll return a Promise.
  * //-
- * table.insert(entries)
+ * table.insert(entries).then(function() {
+ *   // All requested inserts have been processed.
+ * });
+ */
+Table.prototype.insert = function(entries, callback) {
+  entries = arrify(entries).map(propAssign('method', Mutation.methods.INSERT));
+
+  return this.mutate(entries, callback);
+};
+
+/**
+ * Insert or update rows in your table. It should be noted that gRPC only allows
+ * you to send payloads that are less than or equal to 4MB. If you're inserting
+ * more than that you may need to send smaller individual requests.
+ *
+ * Use this method over {bigtable/table#insert} if you want to handle insert
+ * errors as a readable object stream.
+ *
+ * @param {object|object[]} entries - List of entries to be inserted.
+ *     See {module:bigtable/table#mutate}.
+ *
+ * @example
+ * var entries = [
+ *  {
+ *     key: 'alincoln',
+ *     data: {
+ *       follows: {
+ *         gwashington: 1
+ *       }
+ *     }
+ *   }
+ * ];
+ *
+ * table.insertStream(entries)
  *   .on('error', console.error)
- *   .on('complete', function() {
+ *   .on('finish', function() {
```

Comment:
Ah, right. This isn't very stream-y, and I don't think we should call this new method a stream, if I'm understanding it right. A stream would imply to me that I'm getting a writable stream and I can pipe data to it, i.e.

```js
fs.createReadStream('./my-entries.json')
  .pipe(table.createWriteStream())
  .on('error', function(err) {})
  .on('finish', function() {})
```

The way `mutateStream` appears to work isn't as a stream, since all of the data is required up front. The reason we have the event emitter is just to communicate errors as they occur, right? I wonder if we can just keep things simple, and only communicate the errors after the request is complete, and return it in the callback... which I believe is how `mutate` works currently if you give it a callback.

Ideally, we could return a writable stream so the code from my example is possible. We can build the `reqOpts.entries` array progressively as we receive data from a source stream, but it would end up building up memory before we can send the request and flushing it. So... that doesn't work.